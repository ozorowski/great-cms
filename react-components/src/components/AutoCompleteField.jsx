import React from 'react'
import PropTypes from 'prop-types'

import ErrorList from './ErrorList'
import Select from 'react-select'

import './stylesheets/AutoCompleteField.scss'


export default function AutoCompleteField(props){

  const id_for_label = `id_${props.name}`

  function handleChange(values, { action, removedValue }) {
    props.handleChange(values)
  }

  function getLabel() {
    if (props.label) {
      return (
        <label htmlFor={id_for_label} className='great-mvp-field-label'>{props.label}</label>
      )
    }
  }

  return (
    <div>
      {getLabel()}
      <ErrorList errors={props.errors || []} />
      <Select
        id={id_for_label}
        options={props.choices}
        isMulti={true}
        isClearable={true}
        name={props.name}
        onChange={handleChange}
        value={props.value}
        autoFocus={props.autofocus}
        className='great-mvp-autocomplete-field'
        classNamePrefix='great-mvp-autocomplete-field'
        disabled={props.disabled}
        placeholder={props.placeholder}
      />
    </div>
  )
}
  

AutoCompleteField.propTypes = {
  type: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  autofocus: PropTypes.bool,
  errors: PropTypes.array,
}

AutoCompleteField.defaultProps = {
  autofocus: false,
  disabled: false,
  errors: [],
}
