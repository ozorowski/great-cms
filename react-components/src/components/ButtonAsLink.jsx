import React from 'react'
import { node, string } from 'prop-types'
import'./stylesheets/ButtonAsLink.scss'

const ButtonAsLink = ({ children, location }) => (
    <a className="buttonAsLink" href={location}>{children}</a>
)

ButtonAsLink.propTypes = {
    children: node.isRequired,
    location: string.isRequired,
}

export default ButtonAsLink
