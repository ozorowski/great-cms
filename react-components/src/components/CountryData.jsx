import React from 'react'
import PropTypes from 'prop-types'
import { slugify } from '../Helpers'


export default class CountryData extends React.Component {

  constructor(props) {
    super(props)
    this.state = {
      //has to come from data TODO
      time: new Date().toLocaleTimeString('en-GB', {timeZone: 'Australia/Lord_Howe'})
    }
    this.handleClick = this.handleClick.bind(this)
  }

  componentDidMount() {
    this.timerID = setInterval(() => {
      this.tick()
    }, 1000)
  }

  componentWillUnmount() {
    clearInterval(this.timerID)
  }

  tick() {
    this.setState({
      time: new Date().toLocaleTimeString('en-GB', {timeZone: this.props.data.export_marketdata.timezone})
    })
  }

  handleClick() {
    this.props.removeCountry(this.props.data)
  }

  render() {
    const easeofdoingbusiness = this.props.data.easeofdoingbusiness
    const corruption_perceptions_index = this.props.data.corruption_perceptions_index
    const lastyear_import_data = this.props.data.last_year_data
    const export_duty = this.props.data.export_duty
    const sectionID = `export-market-data-${slugify(this.props.data.country)}`

    const utz_offset = '+1030' //Fix me
    const timezone = 'Australia/Lord_Howe' //Fix me
    const rules_regulations = {commodity_name: 'Gin'} //this.props.data.rules_regulations

    const countryData = (
      <>
      <section id={sectionID}>
          <h2 className="h-l p-b-0 inline-block">{ this.props.data.country }  </h2>
          <button
            onClick={this.handleClick}
            id="remove-country"
            className="remove-country-button">
            Remove
            <span className="visually-hidden">{ this.props.data.country }</span>
          </button>
          <div className="flex-grid">
              <div className="c-1-3" id="export-market-data-ease-of-doing-business-rank">
                  <figure className="statistic">
                      <figcaption>
                          <p className="statistic__caption">Ease of doing business rank</p>
                      </figcaption>
                      <p className="statistic__figure">
                          {
                            (easeofdoingbusiness.year_2019) ? (
                              <>
                                <span>{easeofdoingbusiness.year_2019}</span>
                                <span className="statistic__details">out of { easeofdoingbusiness.total }</span>
                              </>
                            ) : ( 'No data' )
                          }
                      </p>
                  </figure>
              </div>
              <div className="c-1-3" id="export-market-data-corruption-perception-index">
                  <figure className="statistic">
                      <figcaption>
                          <p className="statistic__caption">Corruption Perception Index</p>
                      </figcaption>
                      <p className="statistic__figure">{
                          (corruption_perceptions_index.rank) ? (
                            <>
                              { corruption_perceptions_index.rank }
                            </>
                          ) : ( 'No data' ) }</p>
                  </figure>
              </div>
              <div className="c-1-3" id="export-market-data-local-time">
                  <figure className="statistic">
                      <figcaption>
                          <p className="statistic__caption">Local time</p>
                      </figcaption>
                      <p className="statistic__figure">
                          <time>{ this.state.time }</time>
                          <span className="statistic__details">GMT { utz_offset } { timezone }</span>
                      </p>
                  </figure>
              </div>
          </div>
          <div className="flex-grid">
              <div className="c-1-3" id="export-market-data-duty">
                  <figure className="statistic">
                      <figcaption>
                          <p className="statistic__caption">Duty</p>
                      </figcaption>
                      <p className="statistic__figure">
                        {
                          (export_duty > 0) ? (
                            <>
                              { export_duty }%
                            </>
                          ) : ( 'No duty' )
                        }
                      </p>
                  </figure>
              </div>
              <div className="c-1-3" id="export-market-data-import-value">
                {
                  (lastyear_import_data) ? (
                    <figure className="statistic">
                      <figcaption>
                          <p className="statistic__caption">
                              <span className="text-flag-red">{ rules_regulations.commodity_name }</span> import value in { lastyear_import_data.year }
                          </p>
                      </figcaption>
                      <p className="statistic__figure">
                          { lastyear_import_data.trade_value } USD
                          <span className="statistic__details">Source: Comtrade</span>
                      </p>
                  </figure>
                ) : null
              }
              </div>
              <div className="c-1-3" id="export-market-data-year-to-year-change">
                {
                  (lastyear_import_data) ? (
                    <figure className="statistic">
                        <figcaption>
                            <p className="statistic__caption">Year-to-year change</p>
                        </figcaption>
                        <p className="statistic__figure">
                            +{ lastyear_import_data.year_on_year_change }%
                        </p>
                    </figure>
                  ) : null
                }
              </div>
          </div>
      </section>
      <hr/>
      </>
    )

    return (
      <>
        {countryData}
      </>
    )
  }

}

CountryData.propTypes = {
  data: PropTypes.object.isRequired,
}

CountryData.defaultProps = {
  data: {}
}
