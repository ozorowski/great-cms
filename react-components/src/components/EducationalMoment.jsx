import React from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import Tooltip from './Tooltip';

export default class EducationalMoment extends React.Component {
    constructor(props, container) {
        super(props);

        this.state = {
            displayed: this.props.displayed,
            id: null,
            heading: null,
            body: [],
            headingPriority: null
        }

        this.handleHover = this.handleHover.bind(this);
        this.handleFocus = this.handleFocus.bind(this);
        this.handleBlur = this.handleBlur.bind(this);
    }

    componentDidMount() {
        this.extractContent(ReactDOM.findDOMNode(this).parentNode);
    }

    extractContent(container) {
        const id = container.dataset.tooltipId;
        const heading = container.dataset.tooltipHeading;
        const headingPriority = container.dataset.headingPriority;
        const body = container.dataset.tooltipBody.split(';;');

        this.setState({
            id: id,
            heading: heading,
            body: body,
            headingPriority: headingPriority
        });
    }

    handleHover(e) {
        let isDisplayed = e.type === 'mouseenter';

        this.setState({ displayed: isDisplayed });
    }

    handleFocus(e) {
        this.setState({ displayed: true });
    }

    handleBlur(e) {
        this.setState({ displayed: false });
    }

    render() {
        return (
            <div className="wqe"
                onMouseEnter={this.handleHover}
                onMouseLeave={this.handleHover}
                onFocus={this.handleFocus}
                onBlur={this.handleBlur}>
                <button aria-describedby={this.state.id}>Show tooltip</button>

                <Tooltip
                    id={this.state.id}
                    className={this.state.displayed ? '' : 'hidden'}
                    heading={this.state.heading}
                    body={this.state.body}
                />
            </div >
        )
    }

}

EducationalMoment.propTypes = {
    displayed: PropTypes.bool,
    id: PropTypes.string,
    heading: PropTypes.string,
    body: PropTypes.array,
    headingPriority: PropTypes.string
}