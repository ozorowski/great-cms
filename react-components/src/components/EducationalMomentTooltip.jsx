import React from 'react';
import PropTypes from 'prop-types';
import Tooltip from './Tooltip/Tooltip';
import EducationalMomentIcon from './EducationalMomentIcon/EducationalMomentIcon';

/**
 * Dynamically loaded component from HTML fixture
 * 
 * @class EducationalMomentTooltip
 * @selector ".educational-moment--tooltip"
 * 
 * HTML Fixture:
 * 
    <div class="educational-moment educational-moment--tooltip" data-tooltip-type="LEFT">
        <div class="js-hidden educational-moment__tooltip" id="tooltip-id">
            <h3 class="tooltip__heading h-m">Optional heading</h3>
            <p class="tooltip__paragraph">First paragraph</p>
            <p class="tooltip__paragraph">Second paragraph</p>
        </div>
    </div>
 *
 * Inner HTML content is replaced by React if javascript is on
 * However, we persere the originla fixture via this.props.htmlFixture
 * 
 * To switch the view of tooltip use optional parameter on the container:
 * data-tooltip-type="RIGHT"
 */
export default class EducationalMomentTooltip extends React.Component {
    constructor(props) {
        super(props);

        //tooltip type constants
        this.TOOLTIP_TYPE = { Left: 'LEFT', Right: 'RIGHT' };

        this.state = {
            id: null,
            displayed: false,
            loaded: false,
            tooltipContent: null,
            type: this.TOOLTIP_TYPE.Left
        }

        this.bindEvents();
    }

    bindEvents() {
        this.handleHover = this.handleHover.bind(this);
        this.handleFocus = this.handleFocus.bind(this);
        this.handleBlur = this.handleBlur.bind(this);
        this.handleKeyDown = this.handleKeyDown.bind(this);
    }

    componentDidMount() {
        this.extractFixture();
        this.determineType();
        this.addEvents();
    }

    addEvents() {
        document.addEventListener('keydown', this.handleKeyDown, false);
    }

    determineType() {
        let type = this.props.htmlFixture.dataset.tooltipType;
        if (type && type === this.TOOLTIP_TYPE.Right) {
            this.setState({ type: this.TOOLTIP_TYPE.Right });
        }
    }

    extractFixture() {
        let tooltipContent = this.props.htmlFixture.querySelector('.educational-moment__tooltip');
        tooltipContent.classList.remove('js-hidden');

        this.setState({
            id: tooltipContent.id,
            loaded: true,
            tooltipContent: tooltipContent
        });
    }

    handleKeyDown(e) {
        // if ESC button is pressed down make sure tooltip is closed
        // helpful for Zoomtext and other magnifiers
        if (e.keyCode === 27 && this.state.displayed) {
            this.hideTooltip();
        }
    }

    handleHover(e) {
        if (e.type === 'mouseenter') {
            this.showTooltip();
        } else {
            this.hideTooltip();
        }
    }

    handleFocus(e) {
        this.showTooltip();
    }

    handleBlur(e) {
        this.hideTooltip();
    }

    showTooltip() {
        this.setState({ displayed: true });
    }

    hideTooltip() {
        this.setState({ displayed: false });
    }

    render() {
        return (
            <div onMouseEnter={this.handleHover}
                onMouseLeave={this.handleHover}
                onFocus={this.handleFocus}
                onBlur={this.handleBlur}>
                <EducationalMomentIcon hiddenText="Additional information" ariaDescribedBy={this.state.id} />
                {this.state.loaded ?
                    <Tooltip
                        className={`${!this.state.displayed ? 'hidden' : ''} ${(this.state.type === this.TOOLTIP_TYPE.Right) ? 'tooltip--right-side' : ''}`}
                        id={this.state.id}
                        tooltipContent={this.state.tooltipContent} />
                    : ''
                }
            </div>
        )
    }
}

EducationalMomentTooltip.propTypes = {
    displayed: PropTypes.bool,
    id: PropTypes.string,
    loaded: PropTypes.bool,
    tooltipContent: PropTypes.node,
    type: PropTypes.string
}