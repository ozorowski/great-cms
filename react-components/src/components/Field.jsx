import React from 'react'
import PropTypes from 'prop-types'

import ErrorList from './ErrorList'

import './stylesheets/Field.scss'


export default function Field(props){

  const id_for_label = `id_${props.name}`

  function handleChange(event) {
    props.handleChange(event.target.value)
  }

  function getLabel() {
    if (props.label) {
      return (
        <label htmlFor={id_for_label} className="great-mvp-field-label">{props.label}</label>
      )
    }
  }

  return (
    <div>
      {getLabel()}
      <ErrorList errors={props.errors || []} />
      <input
        autoFocus={props.autofocus}
        className="great-mvp-field-input form-control"
        disabled={props.disabled}
        id={id_for_label}
        name={props.name}
        onChange={handleChange}
        placeholder={props.placeholder}
        type={props.type}
        value={props.value}
      />
    </div>
  )
}


Field.propTypes = {
  type: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  autofocus: PropTypes.bool,
  errors: PropTypes.array,
}

Field.defaultProps = {
  autofocus: false,
  disabled: false,
  errors: [],
}
