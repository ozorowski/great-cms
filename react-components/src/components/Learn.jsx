import React from 'react'
import ReactDOM from 'react-dom'
import ButtonAsLink from './ButtonAsLink'
import './stylesheets/Learn.scss'

const Learn = () => (
    <div className="learn-wrapper">
        <h1 className="h1">Start exporting today</h1>
        <p className="paragraph">Start your exporting journey today, and in no time, youâ€™ll have your realistic business plan for selling overseas.</p>
        <ButtonAsLink location="/learn/learn-to-export-step-1">Learn how to export</ButtonAsLink>
    </div>
);

export { Learn }
export default () => ReactDOM.render(<Learn />, document.getElementById('learn-root'))
