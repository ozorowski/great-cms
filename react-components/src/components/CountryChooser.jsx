import React from 'react'
import ReactDOM from 'react-dom'
import PropTypes from 'prop-types'
import Select from 'react-select'
import ErrorList from '@src/components/ErrorList'
import CountryData from '@src/components/CountryData'
import Services from '../Services'
import { slugify } from '../Helpers'
import './stylesheets/CountryChooser.scss'

class CountryChooser extends React.Component {
  constructor(props) {
    super(props)

    const { selectedCountries } = this.props
    const updatedCountryList = this.updatedCountryList(selectedCountries)

    this.state = {
      open: false,
      loading: false,
      selectedCountries,
      selectedCountry: null,
      countryList: updatedCountryList,
      errors: {},
    }

    this.showCountrySelect = this.showCountrySelect.bind(this)
    this.addCountry = this.addCountry.bind(this)
    this.handleGetCountryDataSuccess = this.handleGetCountryDataSuccess.bind(this)
    this.handleGetCountryDataError = this.handleGetCountryDataError.bind(this)
  }

  removeCountry = (country) => {
    const { selectedCountries } = this.state
    const updatedSelectedCountries = selectedCountries.filter((item) => item !== country)
    const updatedCountryList = this.updatedCountryList(updatedSelectedCountries)
    this.setState({ selectedCountries: updatedSelectedCountries, countryList: updatedCountryList })
    return false
  }

  changeCountry = (country) => {
    this.setState({ selectedCountry: country })
  }

  updatedCountryList = (selectedCountries) => {
    const { countryList } = this.props
    return countryList.filter(
      (country) => selectedCountries.filter((selectedCountry) => selectedCountry.country === country.value).length === 0
    )
  }

  addCountry() {
    const { selectedCountry } = this.state
    this.setState({
      loading: true,
      errors: {},
    })

    Services.getCountryData(selectedCountry.label)
      .then(this.handleGetCountryDataSuccess)
      .catch(this.handleGetCountryDataError)
  }

  handleGetCountryDataSuccess(data) {
    const { selectedCountries } = this.state
    const updatedSelectedCountries = selectedCountries.concat(data.target_markets)
    const updatedCountryList = this.updatedCountryList(updatedSelectedCountries)

    this.setState({
      errors: {},
      loading: false,
      selectedCountries: updatedSelectedCountries,
      countryList: updatedCountryList,
    })
  }

  handleGetCountryDataError(errors) {
    this.setState({
      errors: errors.message || errors,
      loading: false,
    })
  }

  showCountrySelect() {
    const { open } = this.state
    this.setState({ open: !open })
  }

  render() {
    const { selectedCountry, loading, open, selectedCountries, errors } = this.state
    const { countryList } = this.state
    let saveButton
    if (selectedCountry) {
      saveButton = (
        <button
          type="button"
          className="country-chooser-save-button"
          id="country-chooser-save-button"
          onClick={this.addCountry}
          disabled={loading}
        >
          Add
        </button>
      )
    }

    let countryInput
    if (open) {
      countryInput = (
        <>
          <div className="country-autocomplete-container m-t-s">
            <Select
              id="country-autocomplete"
              options={countryList}
              isMulti={false}
              isClearable
              disabled={loading}
              name="country"
              value={selectedCountry}
              onChange={this.changeCountry}
              autoFocus
              className="country-autocomplete"
              classNamePrefix="country-autocomplete"
              placeholder="Start typing a country name"
            />
            {saveButton}
          </div>
        </>
      )
    }

    let loadingMessage
    if (loading) {
      loadingMessage = (
        <p className="loading-message">
          Fetching country data
          <span>.</span>
          <span>.</span>
          <span>.</span>
        </p>
      )
    }

    return (
      <>
        {selectedCountries.map((country) => (
          <CountryData data={country} key={slugify(country.country)} removeCountry={this.removeCountry} />
        ))}
        {loadingMessage}
        <div>
          <ErrorList errors={errors.__all__ || []} className="m-v-s" />
        </div>
        <div className={`country-chooser ${open ? 'open' : ''}`}>
          <span className="button--plus" />
          <button
            type="button"
            className="country-chooser-button text-grey bg-stone-90 font-brand bg-hover-stone border-hover-stone pill"
            id="country-chooser-button"
            onClick={this.showCountrySelect}
          >
            Add a country
          </button>
          {countryInput}
        </div>
      </>
    )
  }
}

function createCountryChooser({ element, ...params }) {
  ReactDOM.render(<CountryChooser {...params} />, element)
}

CountryChooser.propTypes = {
  selectedCountries: PropTypes.arrayOf(
    PropTypes.shape({
      export_duty: PropTypes.number.isRequired,
      country: PropTypes.string.isRequired,
      last_year_data: PropTypes.shape({
        year: PropTypes.string.isRequired,
        trade_value: PropTypes.string.isRequired,
        country_name: PropTypes.string.isRequired,
        year_on_year_change: PropTypes.string.isRequired,
      }),
      corruption_perceptions_index: PropTypes.shape({
        rank: PropTypes.number,
        country_code: PropTypes.string.isRequired,
        country_name: PropTypes.string.isRequired,
        cpi_score_2019: PropTypes.number.isRequired,
      }).isRequired,
      easeofdoingbusiness: PropTypes.shape({
        total: PropTypes.number.isRequired,
        year_2019: PropTypes.number,
        country_code: PropTypes.string.isRequired,
        country_name: PropTypes.string.isRequired,
      }).isRequired,
    }).isRequired
  ).isRequired,
  countryList: PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.string.isRequired,
      label: PropTypes.string.isRequired,
    })
  ).isRequired,
}

export { CountryChooser, createCountryChooser }
