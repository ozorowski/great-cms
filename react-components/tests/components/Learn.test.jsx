import React from 'react'
import { shallow } from 'enzyme'
import { Learn } from '@src/components/Learn'

describe('Learn component', () => {
    test('should render proper markup', () => {
        const wrapper = shallow(<Learn />)

        expect(wrapper).toMatchSnapshot()
    })
})
