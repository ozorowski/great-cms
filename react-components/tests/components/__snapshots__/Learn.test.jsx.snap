// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Learn component should render proper markup 1`] = `ShallowWrapper {}`;
